skbeam/core/recip.py

```diff
@@ -342,3 +342,37 @@ def gisaxs(incident_beam, reflected_beam, pixel_size, detector_size,
     qr = np.sqrt(qx**2 + qy**2)
 
     return gisaxs_output(alpha_i, theta_f, alpha_f, tilt_angle, qx, qy, qz, qr)
+
+
+def generate_q_bins(rmax, qmax, pixel_size, distance, wavelength):
+    """
+    Generate the Q bins at the resolution of the detector
+    Parameters
+    -----------
+    rmax: float
+        The maximum radial distance on the detector
```

Comment:
Just not that this is in 'real' units and not pixel count.

Would it make more sense to take in pixel number given that you also take in pixel size and it is easier to get the pixel shape of your detector (my asking the image how big it is) than it is to get the physical size.